crates/services/producer/src/block_producer/tests.rs

```diff
@@ -649,27 +649,33 @@ mod dry_run {
     }
 
     #[tokio::test]
-    async fn dry_run__errors_early_if_height_is_lower_than_chain_tip() {
```

Comment:
I did a quick scan but couldn't see a test for this on the produce or produce_predefined. Maybe I missed it. Do you think we need that coverage?

Reply:
We have test case for `BlockHeightShouldBeHigherThanPrevious` error. 

<img width="735" alt="image" src="https://github.com/user-attachments/assets/df2526b3-a60c-483e-915e-fbcc921dccb1">

But after introduction of the state rewind feature, I don't think that `BlockHeightShouldBeHigherThanPrevious` error makes a lot of sense. Basically importer already is responsible and upgradable executor do this kind of check inside, some maybe we can just remove it from the block producer